File: freight-triage-polyglot-ledger-recovery/environment/intake/src/com/freight/stats/MinMilli.java
package com.freight.stats;

/** min_milli kernel. */
public final class MinMilli implements StatKernel {

    @Override
    public String name() {
        return "min_milli";
    }

    @Override
    public long apply(long[] series) {
        if (series.length == 0) {
            return 0L;
        }
        long best = series[0];
        for (int i = 1; i < series.length; i++) {
            if (series[i] < best) {
                best = series[i];
            }
        }
        return best * 1000L;
    }
}

File: freight-triage-polyglot-ledger-recovery/environment/native/src/stats/min_milli.cpp
#include "freight/stats.h"

#include <algorithm>

namespace freight {

// min_milli kernel.
long long stat_min_milli(const std::vector<long long>& series) {
  if (series.empty()) {
    return 0;
  }
  long long best = series[0];
  for (size_t i = 1; i < series.size(); ++i) {
    if (series[i] < best) {
      best = series[i];
    }
  }
  return best * 1000;
}

}  // namespace freight
